src/Illuminate/Support/Traits/DispatchesActions.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Illuminate\Support\Traits;
+
+use Exception;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Container\Container;
+use Illuminate\Support\Attributes\UnauthorizedActionOverride;
+use ReflectionClass;
+use ReflectionNamedType;
+
+trait DispatchesActions
+{
+    public static function dispatch(mixed ...$params): mixed
+    {
+        $action = static::make(...$params);
+
+        if ($action->authorize()) {
+            return $action->handle();
+        }
+
+        $reflection = new ReflectionClass($action);
+        $attributes = $reflection->getAttributes(UnauthorizedActionOverride::class);
+
+        if (! empty($attributes)) {
```

Comment:
You could avoid the overhead of a function call here as there's only one possible empty state we want to check against:
```suggestion
        if ($attributes === []) {
```

Reply:
Sorry, I accidentally inverted the condition:
```suggestion
        if ($attributes !== []) {
```